syntax = "proto3";

package your_package;

message Emotion {
    string label = 1;
    float probability = 2;
    string emotion_index = 3; 
}

message PredictRequest {
    bytes audio = 1;
    int64 diaryId = 2;
}

message PredictReply {
    repeated Emotion emotions = 1;
    Emotion max_emotion = 2;  
    string stress_index = 3;
}

service MyMLModel {
    rpc Predict(PredictRequest) returns (PredictReply) {}
}

